paddle/fluid/operators/interpolate_op.cu

```diff
@@ -1027,18 +1035,30 @@ static void Interpolate3DCUDAFwd(const framework::ExecutionContext& ctx,
     out_h = new_size[1];
     out_w = new_size[2];
   } else {
-    float scale;
+    float scale_d;
+    float scale_h;
+    float scale_w;
     auto scale_tensor = ctx.Input<Tensor>("Scale");
     if (scale_tensor != nullptr) {
       auto scale_data = get_new_data_from_tensor<float>(scale_tensor);
-      scale = scale_data[0];
+      if (scale_data.size() > 1) {
+        scale_d = scale_data[0];
+        scale_h = scale_data[1];
+        scale_w = scale_data[2];
+      } else {
+        scale_d = scale_data[0];
+        scale_h = scale_data[0];
+        scale_w = scale_data[0];
+      }
     } else {
-      scale = ctx.Attr<float>("scale");
+      scale_d = ctx.Attr<float>("scale_d");
+      scale_h = ctx.Attr<float>("scale_h");
+      scale_w = ctx.Attr<float>("scale_w");
     }
-    if (scale > 0) {
-      out_d = static_cast<int>(in_d * scale);
-      out_h = static_cast<int>(in_h * scale);
-      out_w = static_cast<int>(in_w * scale);
+    if (scale_d > 0 && scale_h > 0 && scale_w > 0) {
```

Comment:
in order to match the old unittest, can not throw in c++

Reply:
why can't  the test be updated?

Reply:
in order to match the old unittest, can not throw here

Reply:
Usually the test should be updated to reflect code change, not the other way around. 
If there is some peculiar reason that it would not be desirable, please elaborate.

Reply:
like this:
https://github.com/PaddlePaddle/Paddle/blob/8c48c7daecbe8419cb34770f0b178c7e13ade123/python/paddle/fluid/tests/unittests/test_bilinear_interp_op.py#L162

in unittest, We will force a scale value to verify the priority of scale and out_size，Although a priority is also determined on the Python side, it is safer to write a unittest this way. So consider not modifying the unittest

Reply:
seems to be a bug, should error out when scale is 0. 
in this case, the priority rules should not depend on value of the tensor.

Reply:
Added the PADDLE_ENFORCE_EQ, please review again